building/how_to_build_jbosstools_faq.adoc

```diff
@@ -123,64 +66,69 @@ This will remove all knowledge about locally built plugins - including things yo
 
-## How to I skip running  tests? How do I make tests not fail? Or only fail after ALL tests run?
+## How to I skip running tests? How do I make tests not fail? Or only fail after ALL tests run?
 
 To skip running tests, you can use these Maven flags:
+
 * `-Dmaven.test.skip=true` (also skip compilation)
 * `-DskipTests` (recommended)
 
 If your reason for skipping tests is to see if everything can run without being stuck on the first test failure, you might also like these flags:
+
 * `-fae`, `--fail-at-end` : Fail at end of build only
 * `-fn`, `--fail-never` : Never fail the build regardless of result
 
 You can also cause test failures to result in JUnit output without failing the build using these flags:
+
 * `-Dmaven.test.error.ignore=true`
 * `-Dmaven.test.failure.ignore=true`
 
+See also link:build_options.adoc[Maven commandline parameters].
+
 ## How can I debug tests in Eclipse when run from Tycho (with Surefire)?
 
-See <a href="http://www.jboss.org/tools/docs/testing.html">http://www.jboss.org/tools/docs/testing.html</a>
+See link:../debugging/debug_tycho_tests.adoc[Debugging Surefire tests].
 
 ## How do I build docs?
 
-<a href="../build_documentation.adoc">See Building JBoss Tools Documentation</a>.
+See link:build_documentation.adoc[Building documentation].
 
 ## What profiles do I need to build? What Maven properties are useful when building?
 
 Most of the time, you don't need any profiles or -D properties. Here are some profiles and properties you might want to use in special cases.
 
-* `-Pmaximum` : selects the default maximum target platform version instead of the default minimum one. Useful when running tests to verify that your code works against a newer target platform (eg., Eclipse 4.2.2 instead of 4.2.0)
+* `-Pmaximum` : selects the default maximum target platform version instead of the default minimum one. Useful when running tests to verify that your code works against a newer target platform (eg., Eclipse 4.4.2 instead of 4.4.0)
 * `-Dtpc.version` : allows you to pick a specific target platform version from those available in Nexus.
 
-See also '<a href="#how-to-i-skip-running--tests-how-do-i-make-tests-not-fail-or-only-fail-after-all-tests-run">How to I skip running tests? How do I make tests not fail? Or only fail after ALL tests run?</a>' above for test-related properties.
+See also link:build_options.adoc[Maven commandline parameters].
+
+
+## My build is failing due to OutOfMemory or PermGen issues! How do I give Maven more memory?
+
+To configure the amount of memory used by Maven, you can define MVN_OPTS as follows, either in the mvn / mvn.bat script you use to run Maven, or set as global environment variables. Here's how to do so for Fedora, Ubuntu, Windows, OSX.
+
+```bash
+set MAVEN_OPTS=-Xms512m -Xmx1024m -XX:PermSize=128m -XX:MaxPermSize=256m
+```
+
+Note: support for `-XX:PermSize` and `-XX:MaxPermSize` was removed in JDK 8.0
```

Comment:
Nope. It was removed. Ask your JDK:

$➔ /opt/jdk1.8.0/bin/java -Xms128M -Xmx256M -XX:PermSize=128M -XX:MaxPermSize=256M
Java HotSpot(TM) 64-Bit Server VM warning: ignoring option PermSize=128M; support was removed in 8.0
Java HotSpot(TM) 64-Bit Server VM warning: ignoring option MaxPermSize=256M; support was removed in 8.0